crates/biome_fs/src/fs.rs

```diff
@@ -100,17 +100,24 @@ pub trait FileSystem: Send + Sync + RefUnwindSafe {
     /// [Self::path_is_symlink()] and this method to return `true` for the same
     /// path.
     fn path_is_dir(&self, path: &Utf8Path) -> bool {
-        Self::path_kind(self, path).is_ok_and(|kind| matches!(kind, PathKind::Directory { .. }))
+        Self::path_kind(self, path).is_ok_and(PathKind::is_dir)
     }
 
     /// Checks if the given path is a symlink
     fn path_is_symlink(&self, path: &Utf8Path) -> bool {
-        Self::path_kind(self, path).is_ok_and(PathKind::is_symlink)
+        Self::symlink_path_kind(self, path).is_ok_and(PathKind::is_symlink)
     }
 
     /// Returns metadata about the path.
+    ///
+    /// This method follows symlinks.
     fn path_kind(&self, path: &Utf8Path) -> Result<PathKind, FileSystemDiagnostic>;
 
+    /// Returns metadata about the path without following symlinks.
+    ///
+    /// In other words, it returns the kind of the symlink itself, if it is one.
```

Comment:
Can you document when it errors? Because based on the documentation, it should return `Result<Option>>`, but maybe it errors if it isn't a symbol link. It's just unclear